main.go

```diff
@@ -209,6 +247,162 @@ func GetDfoActiveContactList(apiUrl string) []byte {
+func GetDfoAuthToken(apiUrl string) DfoAuthTokenObj {
+	contentType := "application/json"
+	var dfoAuthTokenBody DfoAuthTokenBody
+	dfoAuthTokenBody.GrantType = "client_credentials"
+
+	clientId := GetTenantData(clientIdRequest)
+
+	//Validate a valid ClientId was input (just checks string length)
+	validClientId := false
+	for !validClientId {
+		validClientId = Validate(clientId, false, validClientId)
+		if validClientId {
+			dfoAuthTokenBody.ClientId = clientId
+			break
+		} else {
+			clientId = GetTenantData(clientIdRequest)
+		}
+	}
+
+	clientSecret := GetTenantData(clientSecretRequest)
+
+	//Validate a valid ClientSecret was input (just checks string length)
+	validClientSecret := false
+	for !validClientSecret {
+		validClientSecret = Validate(clientSecret, false, validClientSecret)
+		if validClientSecret {
+			dfoAuthTokenBody.ClientSecret = clientSecret
+			break
+		} else {
+			clientSecret = GetTenantData(clientSecretRequest)
+		}
+	}
+
+	bodyJson, _ := json.Marshal(dfoAuthTokenBody)
+	reader := bytes.NewReader(bodyJson)
+	var responseData []byte
+
+	response, err := http.Post(apiUrl, contentType, reader)
+
+	if err != nil {
+		fmt.Print(err.Error())
+		os.Exit(1)
+	}
+
+	if response.StatusCode == 200 {
+		responseData, err = ioutil.ReadAll(response.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	var dfoAuthTokenObj DfoAuthTokenObj
+	if responseData != nil {
+		err := json.Unmarshal(responseData, &dfoAuthTokenObj)
+		if err != nil {
+			fmt.Println("Cannot unmarshal dfoAuthToken")
+		}
+	} else {
+		fmt.Println("DfoAuthToken was null or empty.")
+	}
+	return dfoAuthTokenObj
+}
+
+// GetDeltaList loops through the known contacts from DMW and compares them to the active contacts in DFO and creates a list of contacts that need to be updated
+func GetDeltaList(dmwKnownContacts DmwKnownContacts, dfoActiveContacts DfoActiveContacts) []byte {
+	found := false
+	var newState int32
+	var updateDateStr string
+	type DmwContacts []DmwKnownContact
+	deltaArray := DmwContacts{}
+	var deltaList DmwKnownContacts
+
+	// Loop through DmwKnownContacts and check each contact data in DfoActiveContacts to see if we find a match
+	for _, contact := range dmwKnownContacts.Contacts {
+		for _, data := range dfoActiveContacts.Data {
+			dataId, _ := strconv.ParseInt(data.Id, 10, 64)
+			if contact.ContactID == dataId {
+				found = true
+				// If contact found and states do not match, then we need to update state.
+				var agentUserID OwnerAssigneeUser
+				realContactState := determineContactStateFromData(agentUserID.IncontactId, data.RoutingQueueId, data.Status)
+				if contact.CurrentContactState != int32(realContactState) {
+					newState = int32(realContactState)
+					updateDateStr = data.StatusUpdatedAt
+				}
+				break
+			}
+		}
+
+		// If no match is found or state needs to be updated, add contact data to deltaList
+		if !found {
```

Comment:
This condition will never be true after the first contact is found (found is never set back to false). Maybe move line 316 to just after 324 (initialie found to false at the top of the outer loop).

Reply:
I have revised this.